programs/bpf_loader/src/lib.rs

```diff
@@ -1473,6 +1473,24 @@ pub fn execute<'a, 'b: 'a>(
                 Err(Box::new(error) as Box<dyn std::error::Error>)
             }
             ProgramResult::Err(mut error) => {
+                if invoke_context
+                    .get_feature_set()
+                    .is_active(&solana_feature_set::apply_cost_tracker_during_replay::id())
+                    && !matches!(error, EbpfError::SyscallError(_))
```

Comment:
why the exception for syscall errors?

Reply:
That is in the SIMD. Rational is that syscalls always interrupt a basic block so CU metering is required regardless.

Reply:
ok. I left a comment in the SIMD PR